### ISSUE TITLE
STV catch-up not working 

### ISSUE THREAD

**Issue by chrissxuk** [Catch-up-TV-and-More/plugin.video.catchuptvandmore#913]
When trying to open a show in STV I get a playback error. Think was working yesterday. I have spoken to someone else who has also confirmed this.

**Comment by joaopa00** [comment]
Looks like it is geoblocked, doesn't it?

**Comment by chrissxuk** [comment]
Yes but I use SmartDNS that unblocked it previously and just in case that
wasn't working also tried Surfshark VPN that also unblocked it previously.
Odd neither work now.

Get BlueMail for Android <https://bluemail.me>

**Comment by chrissxuk** [comment]
Also I just checked the STV app and this works perfectly using my SmartDNS
to circumnavigate the geo blocking so I don't think that is the problem.

Get BlueMail for Android <https://bluemail.me>

**Comment by berkhornet** [comment]
I'm having the same issue - extract from kodi log:

2022-07-13 13:43:58.056 T:25844   ERROR <general>: [Catch-up-TV-&-More.support] 404 Client Error: Not Found for url: https://edge.api.brightcove.com/playback/v1/accounts/6204867266001/videos/6299493090001
                                                   Traceback (most recent call last):
                                                     File "C:\Users\mickr\AppData\Roaming\Kodi\addons\script.module.codequick\lib\codequick\support.py", line 306, in run_callback
                                                       redirect = parent_ins(route, arg_params, self.callback_params)
                                                     File "C:\Users\mickr\AppData\Roaming\Kodi\addons\script.module.codequick\lib\codequick\resolver.py", line 67, in __call__
                                                       results = super(Resolver, self).__call__(route, args, kwargs)
                                                     File "C:\Users\mickr\AppData\Roaming\Kodi\addons\script.module.codequick\lib\codequick\script.py", line 169, in __call__
                                                       return route.function(self, *args, **kwargs)
                                                     File "C:\Users\mickr\AppData\Roaming\Kodi\addons\plugin.video.catchuptvandmore\resources\lib\channels\uk\stv.py", line 167, in get_video_url
                                                       item.path = resolver_proxy.get_brightcove_video_json(
                                                     File "C:\Users\mickr\AppData\Roaming\Kodi\addons\plugin.video.catchuptvandmore\resources\lib\resolver_proxy.py", line 206, in get_brightcove_video_json
                                                       resp = urlquick.get(
                                                     File "C:\Users\mickr\AppData\Roaming\Kodi\addons\script.module.codequick\lib\urlquick.py", line 493, in get
                                                       return request('get', url, params=params, **kwargs)
                                                     File "C:\Users\mickr\AppData\Roaming\Kodi\addons\script.module.codequick\lib\urlquick.py", line 487, in request
                                                       return s.request(method=method, url=url, **kwargs)
                                                     File "C:\Users\mickr\AppData\Roaming\Kodi\addons\script.module.codequick\lib\urlquick.py", line 442, in request
                                                       self._raise_for_status(response, raise_for_status)
                                                     File "C:\Users\mickr\AppData\Roaming\Kodi\addons\script.module.codequick\lib\urlquick.py", line 415, in _raise_for_status
                                                       response.raise_for_status()
                                                     File "C:\Users\mickr\AppData\Roaming\Kodi\addons\script.module.requests\lib\requests\models.py", line 960, in raise_for_status
                                                       raise HTTPError(http_error_msg, response=self)
                                                   requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://edge.api.brightcove.com/playback/v1/accounts/6204867266001/videos/6299493090001
2022-07-13 13:43:58.060 T:25844   ERROR <general>: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                                    - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                                   Error Type: <class 'AttributeError'>
                                                   Error Contents: 'HTTPError' object has no attribute 'code'
                                                   Traceback (most recent call last):
                                                     File "C:\Users\mickr\AppData\Roaming\Kodi\addons\plugin.video.catchuptvandmore\addon.py", line 23, in <module>
                                                       main()
                                                     File "C:\Users\mickr\AppData\Roaming\Kodi\addons\plugin.video.catchuptvandmore\addon.py", line 19, in main
                                                       main.error_handler(exception)
                                                     File "C:\Users\mickr\AppData\Roaming\Kodi\addons\plugin.video.catchuptvandmore\resources\lib\main.py", line 290, in error_handler
                                                       code = exception.code
                                                   AttributeError: 'HTTPError' object has no attribute 'code'
                                                   -->End of Python script error report<--
                                                   
2022-07-13 13:43:58.335 T:13872   ERROR <general>: Playlist Player: skipping unplayable item: 0, path [plugin://plugin.video.catchuptvandmore/resources/lib/channels/uk/stv/get_video_url/?_pickle_=800495bb000000000000007d94288c075f7469746c655f948c2154686520497063726573732046696c65202d20457069736f64652031206f662036948c076974656d5f6964948c03737476948c08766964656f5f6964948a06d1420cb7ba058c087375627469746c65948c5768747470733a2f2f66696c65732e7374762e74762f706c617965722f7375627469746c65732f7468652d697063726573732d66696c652d7365726965732d312d657069736f64652d312d6f662d362e737562732e76747494752e]
2022-07-13 13:43:58.336 T:13872    INFO <general>: Loading skin file: DialogConfirm.xml, load type: KEEP_IN_MEMORY

**Comment by berkhornet** [comment]
I wonder if an account with STV is required - I needed to logon to stream via the web site

**Comment by chrissxuk** [comment]
I've now heard from someone else who says this stopped working on Tuesday, they are in the UK.

**Comment by berkhornet** [comment]
I should add that the error I documented previously happened in the UK, and I was able to play streams via a browser successfully. I did need to log on to my STV account to play streams via browser and I  noted there are no account settings with the addon for STV.

**Comment by chrissxuk** [comment]
Surely nothing to do with geo blocking being it isn't working in the UK?

**Comment by ajuniper** [comment]
Same problem here, another UK user where it's stopped working, gives a 404 error.  No clever DNS or VPN.
Running it thru LibreElec on a Pi.  Same error message, similar URL.

**Comment by 9chins** [comment]
Same problem, started last week, I am new to this, will someone read this and fix the problem ?

**Comment by chrissxuk** [comment]
I have emailed one of the developers directly. Hopefully they will assign someone to fix the issue soon.

**Comment by 9chins** [comment]
Cheers, thankyou

**Comment by joaopa00** [comment]
Unfortunately, even when I manage to fecth the identity of the video, I receive this error message from inputstream adaptive

inputstream.adaptive: Unsupported encryption method: SAMPLE-AES

**Comment by maxoutputuk** [comment]
I'm having the same issue with stv catchup I'm in the UK stv live works fine only catchup not working exactly same issue as above 404 not found not sure when it happened only just around a week

**Comment by berkhornet** [comment]
> inputstream.adaptive: Unsupported encryption method: SAMPLE-AES

Looking at the github at https://github.com/xbmc/inputstream.adaptive/tree/Matrix, there seem to be some issues with AES. 

Perhaps the developers need to raise an issue against inputstream.adaptive to progress further.

**Comment by berkhornet** [comment]
> Unfortunately, even when I manage to fecth the identity of the video, I receive this error message from inputstream adaptive


Looking at the github at https://github.com/xbmc/inputstream.adaptive/tree/Matrix, there seem to be some issues with AES. 

Perhaps the developers need to raise an issue against inputstream.adaptive to progress further.

**Comment by joaopa00** [comment]
I had some progress. I managed to play <<murder in Provence 3>>
Yeah!!
Have to manage to fetch the videos now... That's the tricky part.

**Comment by berkhornet** [comment]
Great news, well done! If you need any help testing fixes please let me know.

**Comment by berkhornet** [comment]
Many thanks. Will do. Give me a day or so.

Mick
________________________________

**Comment by chrissxuk** [comment]
I just updated to this test version but I am unable to open the STV within catch-up. I get an error message that says SYNTAX ERROR • POSITIONAL ARGUMENT FOLLOWS KEYWORD ARGUMENT (STV.PY, LINE 156)

**Comment by joaopa00** [comment]
Should work now... Sorry for the trouble.

**Comment by chrissxuk** [comment]
Don't apologise you're doing a great job I will try again

**Comment by chrissxuk** [comment]
Fantastic it works! 

I went into the STV0 most popular section and tested the first 8 and they all played perfectly. 

A few things to note:- 

After updating on first use it told me to install Inputstream Adaptive. Obviously I didn't need that before then.

7 out of the 8 shows I tested played in 720, including Coronation Street which I know was previously 1080 because I always watch that. The show playing 1080 was Sedition - Episode 1 of 6. Odd that.

**Comment by joaopa00** [comment]
>After updating on first use it told me to install Inputstream Adaptive
It is because some videos are drm-protected. To play them, Catchuptvandmore needs Inputstreamadaptive.

I will wait for one day or two. If nobody meets problems, I will push the patch in git.

**Comment by chrissxuk** [comment]
Great work. Shame we've lost the 1080 though, any idea why? It's 1080 on Hub via smart TV with adds v no adds in 720 on Kodi I guess.

**Comment by chrissxuk** [comment]
So why were they before and not now?

**Comment by berkhornet** [comment]
Probably changes at the stv server side of things.

**Comment by joaopa00** [comment]
@chris I tried "Murder in Provence" and it proposes the resolution 1920x1080.
Check the configuration of Catchuptvandmore specially the item <<quality and content>>

**Comment by maxoutputuk** [comment]
> thank you so very much you have made my wife very very happy you are a star

**Comment by 9chins** [comment]
Hi, do I have to uninstall old version or can I overwrite with test version

**Comment by joaopa00** [comment]
Just wait a few minutes. I will push a new beta very soon

**Comment by berkhornet** [comment]
@joaopa00 many thanks for spending time on this bug. all working great now.

**Comment by chrissxuk** [comment]
Thank you for fixing this.

**Comment by chrissxuk** [comment]
With regards to getting 1080p changing the video quality to dialog in settings gave me a menu including 1080p but even when selecting 1080p it played 720p. But I stumbled on a fix. By default in settings use inputstream hls is on but switching it off gives me 1080p. No idea why but it worked. 🙂

**Comment by joaopa00** [comment]
Good news. I am happy you find a solution for your problem.

**Comment by chrissxuk** [comment]
Yesterday with the patch this worked perfectly but today when attempting to open the STV section of catch-up there is an error, this follows it hanging for a long time. This is happening on two devices a Firestick and a Shield. 
![Screenshot_20220802-153415_Gallery](https://user-images.githubusercontent.com/79976943/182388851-d78585c8-6dd8-4df9-801a-8f885a87efdc.jpg)

**Comment by joaopa00** [comment]
Unable to reproduce in Linux neither Android as well with Kodi 18 as Kodi 19.

**Comment by chrissxuk** [comment]
I installed from your test version https://github.com/joaopa00/plugin.video.catchuptvandmore/archive/refs/heads/dev.zip
I am using Kodi 19.
![20220802_165903](https://user-images.githubusercontent.com/79976943/182407232-f5f9e61b-e411-4a89-bf7c-f6b2e3e35989.jpg)

**Comment by chrissxuk** [comment]
All good now. Sorry to have wasted your time as the issue wasn't with this add-on it was with my SmartDNS I use to geo unblock.

**Comment by ajuniper** [comment]
I'm using this fix on a Pi4 using LibreElec 9.2.8 / Kodi 18.9 and found a couple of minor issues, due to LibreElec 9.2.8 still using Python 2.7.16.

In file /storage/.kodi/addons/plugin.video.catchuptvandmore/resources/lib/resolver_proxy.py there are what appear to be debug prints at lines 339 and 350 which attempt to output non ASCII characters, which Python 2.7 does not handle and crashes.  I commented out these lines and all is well, videos play as expected.

**Comment by joaopa00** [comment]
Can you report the errors given by kodi. Please attach the log.

**Comment by joaopa00** [comment]
Unable to reproduce with Kodi 18.9 in Ubuntu. Can you give the steps to reproduce this problem?

**Comment by ajuniper** [comment]
Actual log file appears to have been lost but these were the stack traces:
```
2022-08-14 22:29:26.077 T:2332320576   ERROR: [Catch-up-TV-&-More.support] 'ascii' codec can't encode character u'\xe9' in position 7: ordinal not in range(128)
                                            Traceback (most recent call last):
                                              File "/storage/.kodi/addons/script.module.codequick/lib/codequick/support.py", line 306, in run_callback
                                                redirect = parent_ins(route, arg_params, self.callback_params)
                                              File "/storage/.kodi/addons/script.module.codequick/lib/codequick/resolver.py", line 67, in __call__
                                                results = super(Resolver, self).__call__(route, args, kwargs)
                                              File "/storage/.kodi/addons/script.module.codequick/lib/codequick/script.py", line 169, in __call__
                                                return route.function(self, *args, **kwargs)
                                              File "/storage/.kodi/addons/plugin.video.catchuptvandmore/resources/lib/channels/uk/stv.py", line 156, in get_video_url
                                                return resolver_proxy.get_brightcove_video_json(plugin, data_account, data_player, data_video_id, key, headers, download_mode, subtitle)
                                              File "/storage/.kodi/addons/plugin.video.catchuptvandmore/resources/lib/resolver_proxy.py", line 339, in get_brightcove_video_json
                                                print("identité video = %s" % data_video_id)
                                              File "<string>", line 7, in write
                                            UnicodeEncodeError: 'ascii' codec can't encode character u'\xe9' in position 7: ordinal not in range(128)
```
and
```
2022-08-14 22:39:09.623 T:2870211392   ERROR: [Catch-up-TV-&-More.support] 'ascii' codec can't encode character u'\xe9' in position 802: ordinal not in range(128)
                                            Traceback (most recent call last):
                                              File "/storage/.kodi/addons/script.module.codequick/lib/codequick/support.py", line 306, in run_callback
                                                redirect = parent_ins(route, arg_params, self.callback_params)
                                              File "/storage/.kodi/addons/script.module.codequick/lib/codequick/resolver.py", line 67, in __call__
                                                results = super(Resolver, self).__call__(route, args, kwargs)
                                              File "/storage/.kodi/addons/script.module.codequick/lib/codequick/script.py", line 169, in __call__
                                                return route.function(self, *args, **kwargs)
                                              File "/storage/.kodi/addons/plugin.video.catchuptvandmore/resources/lib/channels/uk/stv.py", line 156, in get_video_url
                                                return resolver_proxy.get_brightcove_video_json(plugin, data_account, data_player, data_video_id, key, headers, download_mode, subtitle)
                                              File "/storage/.kodi/addons/plugin.video.catchuptvandmore/resources/lib/resolver_proxy.py", line 350, in get_brightcove_video_json
                                                print("reponse total = %s" % resp.text)
                                              File "<string>", line 7, in write
                                            UnicodeEncodeError: 'ascii' codec can't encode character u'\xe9' in position 802: ordinal not in range(128)
```

I don't know which Python version is in your Ubuntu but Python 2.7.16 I suspect is the root cause as I've found in the past characters which are not plain ASCII are trouble.
The steps to reproduce were simple - just go to STV player and play content - specific content I tested with was McDonnell and Dodds and Death in Provence.

**Comment by joaopa00** [comment]
Did you install the locale addon?

**Comment by ajuniper** [comment]
Sorry, wasn't aware there was a locale addon so did not knowingly install it, although dependencies might have pulled that in without me knowing.  How do I check?  And/or where do I find it to install?

**Comment by joaopa00** [comment]
apt-get install locales
should be the trick.

**Comment by joaopa00** [comment]
Forget what I wrote above. I have just seen the problem for you.

You use an old version I wrote for testing. That's not the official beta. You have to install it.
Here is the wiki to install the official beta:
[How to install the official beta](https://catch-up-tv-and-more.github.io/installation/)

**Comment by ajuniper** [comment]
Unfortunately there's no package manager on LibreElec... but finding the locale package and installing that does seem to have helped.
I'll install the beta anyway (y)

**Comment by ajuniper** [comment]
Beta all seems good too, thanks for the help!

**Comment by joaopa00** [comment]
Good news. Enjoy STV ....`
